### PR TITLE
coinbaseinternational

### PR DESCRIPTION
**Pull request by pcriadoperez**
- TODO: update clientOrderId prefix
- For pagination coinbasepro uses a limit and offset I used `fetchPaginatedCallIncremental` as it was the current function I saw that fit the best and works in testing, but in the future it may make sense to have a fetchPaginate function specific for this case.

### Tests ran:
- Ran all funcitons in typescript and stored test results in static tests.
- Shadow tested some functions in python and php
- Ran `node run-tests coinbaseinternational --sandbox`

### CODE REVIEW

File: ts/src/coinbaseinternational.ts [hunk 1]
```diff
@@ -0,0 +1,1849 @@
...
+        const datetime = this.safeString (trade, 'event_time');
+        return this.safeTrade ({
+            'info': trade,
+            'id': this.safeString2 (trade, 'fill_id', 'exec_id'),
+            'order': this.safeString (trade, 'order_id'),
+            'timestamp': this.parse8601 (datetime),
+            'datetime': datetime,
+            'symbol': this.safeSymbol (marketId, market),
+            'type': undefined,
+            'side': this.safeString (trade, 'side').toLowerCase (),
```
Comment (carlosmiei) [inline ts/src/coinbaseinternational.ts:785]: we can use safeStringLower here

File: ts/src/coinbaseinternational.ts [hunk 2]
```diff
@@ -0,0 +1,1849 @@
...
+        }
+        const orders = await this.v1PrivateDeleteOrders (this.extend (request, params));
+        return this.parseOrders (orders, market);
+    }
+
+    async editOrder (id: string, symbol: string, type: OrderType, side: OrderSide, amount: number = undefined, price: number = undefined, params = {}) {
+        /**
+         * @method
+         * @name coinbaseinternational#editOrder
+         * @description edit a trade order
```
Comment (carlosmiei) [inline ts/src/coinbaseinternational.ts:1488]: we're missing the doc here https://docs.cloud.coinbase.com/intx/reference/modifyorder

File: ts/src/coinbaseinternational.ts [hunk 3]
```diff
@@ -0,0 +1,1850 @@
...
+                '15m': 'FIFTEEN_MINUTE',
+                '30m': 'THIRTY_MINUTE',
+                '1h': 'ONE_HOUR',
+                '2h': 'TWO_HOUR',
+                '6h': 'SIX_HOUR',
+                '1d': 'ONE_DAY',
+            },
+            'options': {
+                'clientOrderId': 'ccxt',
+                'portfolio': '1wp37qsc-1-0', // default portfolio id
```
[the remark this answers is not in the document]
Comment (pcriadoperez) [reply ts/src/coinbaseinternational.ts:240]: Yes, using `fetchAccounts` you can get the list of portfolios.
Good catch I'll remove it.

File: ts/src/coinbaseinternational.ts [hunk 4]
```diff
@@ -0,0 +1,1850 @@
...
+            'instrument': market['id'],
+            'size': this.amountToPrecision (market['symbol'], amount),
+        };
+        if (stopPrice !== undefined) {
+            if (type === 'limit') {
+                typeId = 'STOP_LIMIT';
+            } else {
+                typeId = 'STOP';
+            }
+            request['stop_price'] = stopPrice;
```
Comment (carlosmiei) [inline ts/src/coinbaseinternational.ts:1264]: can you place a stop order? having some difficulties with it

File: ts/src/coinbaseinternational.ts [hunk 5]
```diff
@@ -0,0 +1,1850 @@
...
+        let market = undefined;
+        if (symbol) {
+            market = this.market (symbol);
+            request['instrument'] = market['id'];
+        }
+        const orders = await this.v1PrivateDeleteOrders (this.extend (request, params));
+        return this.parseOrders (orders, market);
+    }
+
+    async editOrder (id: string, symbol: string, type: OrderType, side: OrderSide, amount: number = undefined, price: number = undefined, params = {}) {
```
Comment (carlosmiei) [inline ts/src/coinbaseinternational.ts:1484]: hum does not seem to be working properly 

```
ExchangeError coinbaseinternational {"title":"io.javalin.http.BadRequestResponse: Replace order rejected (UNKNOWN_ORDER)","status":400}
---------------------------------------------------
[ExchangeError] coinbaseinternational {"title":"io.javalin.http.BadRequestResponse: Replace order rejected (UNKNOWN_ORDER)","status":400}

    at handleErrors               …ers/cjg/Git/ccxt9/ccxt/js/src/coinbaseinternational.js:1805  
    at                            Users/cjg/Git/ccxt9/ccxt/js/src/base/Exchange.js:962          
    at processTicksAndRejections  node:internal/process/task_queues:95                          
    at fetch2                     Users/cjg/Git/ccxt9/ccxt/js/src/base/Exchange.js:3541         
    at request                    Users/cjg/Git/ccxt9/ccxt/js/src/base/Exchange.js:3544         
    at editOrder                  …ers/cjg/Git/ccxt9/ccxt/js/src/coinbaseinternational.js:1489  
    at async run                  Users/cjg/Git/ccxt9/ccxt/examples/js/cli.js:340               

coinbaseinternational {"title":"io.javalin.http.BadRequestResponse: Replace order rejected (UNKNOWN_ORDER)","status":400}
```

File: ts/src/coinbaseinternational.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (carlosmiei) [reply ts/src/coinbaseinternational.ts:240]: @pcriadoperez yeah and we can memoize that value

[the remark this answers is not in the document]
Comment (pcriadoperez) [reply ts/src/coinbaseinternational.ts:240]: By memoize you mean on loadMarkets I should call fetchAccounts and use the first account as default portfolio id?

File: ts/src/coinbaseinternational.ts (hunk 4 above)
Comment (pcriadoperez) [reply ts/src/coinbaseinternational.ts:1264]: It was requiring a to set the timeInForce to GTC. I made it now default to GTC, to make it easier and should work with: 
`cj coinbaseinternational createStopOrder BTC/USDC market buy 0.01 60000 59000 --test`

File: ts/src/coinbaseinternational.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (carlosmiei) [reply ts/src/coinbaseinternational.ts]: @pcriadoperez If that's the default porfolio then yes

**Comment by carlosmiei** [comment]
coinbase-international-resized

**Pull request merged**